pep-0011.txt

```diff
@@ -14,109 +14,116 @@ Post-History: 18-Aug-2007,
-considered supported.
+This PEP also lists what plaforms *are* supported by the CPython
+interpreter. This lets people know what platforms are directly
+supported by the CPython development team.
 
-Unsupporting platforms
-----------------------
 
-If a certain platform that currently has special code in CPython is
-deemed to be without enough Python users or lacks proper support from
-the Python development team and/or a buildbot, a note must be posted
-in this PEP that this platform is no longer actively supported.  This
-note must include:
+Support tiers
+=============
 
-- the name of the system
-- the first release number that does not support this platform
-  anymore, and
-- the first release where the historical support code is actively
-  removed
+Platform support is broken down into *tiers*. Each tier comes with
+different requirements which lead to different promises being made
+about support.
 
-In some cases, it is not possible to identify the specific list of
-systems for which some code is used (e.g. when autoconf tests for
-absence of some feature which is considered present on all
-supported systems).  In this case, the name will give the precise
-condition (usually a preprocessor symbol) that will become
-unsupported.
+Tier 1
+------
 
-At the same time, the CPython source code must be changed to
-produce a build-time error if somebody tries to install Python on
-this platform.  On platforms using autoconf, configure must fail.
-This gives potential users of the platform a chance to step
-forward and offer maintenance.
+- `CI failures <https://github.com/python/cpython/actions/workflows/build.yml?query=branch%3Amain+is%3Acompleted>`__ block releases.
+- Changes which would break the ``main`` branch are not allowed to be merged;
+  any breakage may be reverted immediately.
+- All core developers are responsible to keep these platforms,
+  and thus ``main``, working.
+- Promotion to this tier requires team consensus/SC approval.
+
+====================== =====
+Target Triple          Notes
+====================== =====
+i686-pc-windows-msvc
+x86_64-pc-windows-msvc
+x86_64-apple-darwin    BSD libc, clang
+x86_64-\*-linux-gnu    glibc, gcc
+====================== =====
+
+
+Tier 2
+------
+
+- Must have a reliable buildbot.
+- At least **two** core developers are signed up to support the platform.
+- Changes which break any of these platforms are to be fixed or
+  reverted within 24 hours.
+- Failures on these platforms block a release.
+- Promotion to this tier requires consensus/SC approval.
+
+====================== ========================== ============================================== ========
+Target Triple          Notes                      Buildbot                                       Contacts
+====================== ========================== ============================================== ========
+aarch64-apple-darwin   clang                      https://buildbot.python.org/all/#/builders/725 XXX, XXX
```

Comment:
If `x86_64-apple-darwin` is a Tier 1 platform, then this must be as well. As of today, nearly all Macs being manufactured by Apple are of this architecture (i.e. Apple Silicon) and Apple has made clear that the few remaining legacy Intel-based Macs still being offered will be replaced in the near future (likely by the end of 2022). So, eventually `x86_64-apple-darwin` will fade in importance as older Macs are retired but that will take many years.

Reply:
Until this is available in Github CI so it can be a reliable **merge** blocker I don't believe we have a choice but to list it as Tier 2.  The tier shouldn't be interpreted as importance to the world.  Anything listed in 1 or 2 is already a **release** blocker.

Reply:
_(and agreed, x86_64 and universal Darwin binaries will likely remain important through 2028)_

Reply:
OK, if the lack of GitHub CI support is the hangup at the moment, I agree with that assessment although we should perhaps note that, once it is available, we need to move this to Tier 1.  You can add my name and I assume @ronaldoussoren 's name. Perhaps also @ambv ?

Reply:
You can add my name as well.

Reply:
You can add my name too cc @ronaldoussoren @ned-deily 
My main equipment is m1 MacBook pro now and my whole core dev's experience is based on apple x86 and aarch64, so I think that I am ready to support this platform :)

Reply:
```suggestion
aarch64-apple-darwin   clang                      https://buildbot.python.org/all/#/builders/725 Ned Deily, Ronald Oussoren, Dong-he Na
```

@ned-deily yes, the assumption is as soon as we have CI support for a platform and we are using it the platform will get promoted to tier 1.